app/code/Magento/SalesRule/Model/Rule/DataProvider.php

```diff
@@ -55,13 +62,15 @@ public function __construct(
         CollectionFactory $collectionFactory,
         \Magento\Framework\Registry $registry,
         \Magento\SalesRule\Model\Rule\Metadata\ValueProvider $metadataValueProvider,
+        DataPersistorInterface $dataPersistor,
```

Comment:
Please add new dependencies in a backward compatible way, see "Adding a parameter to Constructor" section of https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#php